### PR TITLE
Feature Hint cards and app walkthrough

### LINKED ISSUE #1180: Visual tips on Kedro-Viz
## Description 
Add visual tips/tutorial on Viz to highlight new features and explain how to use them. 

## Further ideas
- Expand such tips to framework:
   - In viz add more elaborate tips that also explain the functionality on the Framework side, e.g. when clicking on a dataset it could talk through how to use dataset or any new releases
   - On a `pip install kedro` add info about new features

### PR DESCRIPTION
**Pull request by tynandebold**
## Description

Resolves #1180.

## Development notes

I've added what I'm calling the `AppOnboarding` component and necessary files. I'm opening the PR a little early to get it into our hands so we can test it out. I feels suboptimal right now.

## QA notes

https://github.com/kedro-org/kedro-viz/assets/16869061/c29a48ab-2e2e-458a-990b-4fa21105188f

Look at the Gitpod link and play around with it and reference the design [here](https://www.figma.com/file/2WSNDAdG3JZS7qdaiH6Qz6/Kedro-Component-Library?type=design&node-id=102%3A8941&mode=dev).

I think there's a wrinkle or two with the concept itself. See my comment here for more details: https://github.com/kedro-org/kedro-viz/issues/1180#issuecomment-1638452433.

Other things that we can improve:
- The hit area of the small circles that are used to change to another card are much too small. It's too hard to click them.
- There's a lot of "to be decided" language in the design concept that hasn't been decided.
- The small "New feature" text element in the design doesn't make much sense to me, since a lot of what we want to show isn't actually a new feature. We can improve this language.

Still todo from the dev side:

- [ ] Write e2e and unit tests

## Checklist

- [x] Read the [contributing](/CONTRIBUTING.md) guidelines
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [x] Updated the documentation to reflect the code changes
- [ ] Added new entries to the `RELEASE.md` file
- [ ] Added tests to cover my changes

<a href="https://gitpod.io/#https://github.com/kedro-org/kedro-viz/pull/1454"></a>

### CODE REVIEW

**Comment by stephkaiser** [comment]
agree we need some UX + UI improvements here

**Comment by stephkaiser** [comment]
Thanks Tynan! 

Some notes:

1. When opening the metadata panel, the feature hint moves but the highlight circle stays fixed which looks weird in some situations, can we make the circle move with the feature hint too? (see pics below)
Screenshot 2023-08-02 at 16 59 13 
Screenshot 2023-08-02 at 17 00 54


2. When the metadata panel is open and I navigate to the next feature hint, the highlight circle disappears for the first two hints that are in the flowchart, and then re-appears on card 3 for experiment tracking. I assume this is happening because the flowchart doesnt adjust with the highlight circle when the metadata panel is open? - can we make the flowchart move with the circle when the user navigates from card 1 to card 2 with the panel open?

4. Learn More button - can we make this aligned to the text above, theres a bit too much padding on the left

6. Feature hints restart from the first card when I click on experiment tracking and go back to flowchart - could we make it so when the user goes back to the Flowchart view we are on the same current feature hint?

7. Maybe we should remove the '' in the text for the dismiss toast message: _'Settings' panel_. 

9. On second thought, could we remove the last hint "Export visualisation" - I feel like this is maybe more suitable for the future onboarding journey rather than feature hints - what do you guys think? So there will be 5 in total and last hint would be the settings panel.

11. For the Filters and tags hint - could we add a link to docs here too? maybe a link to [how to tag a node](https://docs.kedro.org/en/latest/nodes_and_pipelines/nodes.html#how-to-tag-a-node)?

**Comment by tynandebold** [comment]
Thanks for a thorough review @stephkaiser!

I've addressed all your comments now. Feel free to check things again :)

**Comment by stephkaiser** [comment]
looks awesome! @tynandebold no comments from me :)

**Review by stephkaiser — APPROVED** [review]

**Review by NeroOkwa — APPROVED** [review]
@tynandebold this looks good.

For the first card - If you select the dataset and click **Expand Preview Table**, The page opens showing the datasets but also the yellow indicator as seen below:

Screenshot 2023-08-03 at 16 28 19

This is a similar experience with the second card when you select 'Price Histogram' and select '**Expand Plotly Visualisation**' the yellow indicator is still shown as seen below:

Screenshot 2023-08-03 at 16 28 52

File: src/components/ui/delayed-renderer/delayed-renderer.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+      onExited={() => setIsShown(true)}
+      timeout={300}
+      unmountOnExit
+    >
+      <div>{children}</div>
+    </CSSTransition>
+  );
+};
+
+export default DelayedRenderer;
```
Comment (ravi-kumar-pilla) [inline src/components/ui/delayed-renderer/delayed-renderer.js:31]: loved this !!

**Review by ravi-kumar-pilla — APPROVED** [review]
Awesome work !! I have left few comments, nits and questions. I feel neither of them is a major blocker. Approving the PR. Thank you

**Comment by ravi-kumar-pilla** [comment]
> @tynandebold this looks good.
> 
> For the first card - If you select the dataset and click **Expand Preview Table**, The page opens showing the datasets but also the yellow indicator as seen below:
> 
> Screenshot 2023-08-03 at 16 28 19
> 
> This is a similar experience with the second card when you select 'Price Histogram' and select '**Expand Plotly Visualisation**' the yellow indicator is still shown as seen below:
> 
> Screenshot 2023-08-03 at 16 28 52

Nice catch @NeroOkwa . 

When someone is in the tour, should we disable/not allow the users to interact with the app or make changes ? @tynandebold @stephkaiser

File: src/components/feature-hints/feature-hints-content.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    elementId: '#nodes .pipeline-node--data:first-of-type .pipeline-node__bg',
+  },
+  {
+    title: 'Plotly and Matplotlib',
+    description:
+      'Kedro-Viz supports integration with Plotly and Matplotlib in order to make interactive visualisations of a Kedro project.',
+    learnMoreLink:
+      'https://docs.kedro.org/en/stable/visualisation/visualise_charts_with_plotly.html',
+    elementId: '.pipeline-node__bg--plotly',
+    image:
```
[the remark this answers is not in the document]
Comment (tynandebold) [reply src/components/feature-hints/feature-hints-content.js:17]: Currently we don't have a mechanism that allows us to bundle static images in Viz. We'd need to account for the potential of the path changing in local development versus production, too.

Something to investigate for the future.

File: src/components/feature-hints/feature-hints.js [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+                <CloseIcon />
+              </div>
+            </div>
+            <div className="feature-hints__header">
+              {featureHintsContent[featureHintStep].title}
+            </div>
+            {featureHintsContent[featureHintStep].image && (
+              <img
+                alt={featureHintsContent[featureHintStep].title}
+                src={featureHintsContent[featureHintStep].image}
```
[the remark this answers is not in the document]
Comment (tynandebold) [reply src/components/feature-hints/feature-hints.js:106]: See my comment above about this.

File: src/components/settings-modal/settings-modal.js [hunk 4]
```diff
@@ -106,6 +121,22 @@ const SettingsModal = ({
...
               }}
             />
+            <SettingsModalRow
+              id="showFeatureHints"
+              name={settingsConfig['showFeatureHints'].name}
+              toggleValue={showFeatureHintsValue}
+              description={settingsConfig['showFeatureHints'].description}
+              onToggleChange={(event) => {
+                setShowFeatureHintsValue(event.target.checked);
+                setHasNotInteracted(false);
```
[the remark this answers is not in the document]
Comment (tynandebold) [reply src/components/settings-modal/settings-modal.js:131]: This is the state variable that changes the submit button from disabled to active.

File: src/components/feature-hints/feature-hints.scss [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+    display: flex;
+    justify-content: space-between;
+
+    a,
+    .button {
+      flex-shrink: 0;
+    }
+
+    .button button {
+      color: colors.$black-900;
```
[the remark this answers is not in the document]
Comment (tynandebold) [reply src/components/feature-hints/feature-hints.scss:72]: I don't think we use it consistently. String interpolation of this kind is only needed when you want to use the variable somewhere it isn't allowed.

**Pull request merged**